tests/benchmark/scripts/json_reader.py

```diff
@@ -0,0 +1,212 @@
+        # dict that maps the experiments to the set of parameters it was performed with
+        self.__families = dict()
+        # a dictionary of all measurements, maps the name of an experiment to the attributes of interest
+        self.__measurements = dict()
+        # attributes of interest in each measurement
+        self.__attributes = {
+            "real_time": {"pretty_name": "Real Time", "unit": "ms"},
+            "cpu_time": {"pretty_name": "CPU Time", "unit": "ms"},
+            "max_bytes_used": {"pretty_name": "Maximum Bytes Used", "unit": "byte"},
+            "total_allocated_bytes": {
+                "pretty_name": "Total Allocated Bytes",
+                "unit": "byte",
+            },
+        }
+
+        self.__load_data(file_names)
+        self.__lookup_matches()
+        self.__lookup_families()
+        self.__store_measurements()
+
+    def __load_data(self, file_names):
+        """
+        Loads json objects into a list of py dictionaries
+        """
+        for i, file_name in enumerate(file_names):
+            log.info("Loading file {} : {}".format(i + 1, file_name))
+            with open(file_name, "r") as f:
+                self.__data.append(json.load(f))
+            f.close()
+            self.__ids.append(FileMeta(file_name, "benchmark_" + format(i + 1, "02d")))
+            log.info(
+                "Loaded {}, pretty name = {}".format(
+                    self.__ids[-1].path, self.__ids[-1].pretty_name
+                )
+            )
+        log.info("All files loaded")
+        if len(self.__data) > 1:
+            self.__has_multiple_contenders = True
+        else:
+            log.info("Only a baseline measurement is available")
+
+    def __lookup_matches(self):
+        """
+        Finds benchmarks with matching names in all input files
+        """
+        log.info("Looking up matching measurements")
+        # create a list of lists of benchmark names
+        names = list()
+        for data in self.__data:
+            node = data["benchmarks"]
+            name = list()
+            for child in node:
+                name.append(child["name"])
+            names.append(name)
+
+        # filter the matches
+        if len(names) == 1:
+            self.__matches = names[0]
+        else:
+            for i in range(1, len(names)):
+                self.__matches = [item for item in names[i - 1] if item in names[i]]
+            if self.__matches:
+                log.info("Found matches: {}".format(self.__matches))
+            else:
+                log.error("Failed to find matches in all parsed json files")
+                sys.exit()
+
+    sep = "/"
+
+    def __lookup_families(self):
+        """
+        Looks up families of measurements in matching measurements
+        """
+        log.info("Looking up families of measurements")
+        # get unique prefixes (must preserve ordering)
+        match_prefixes = list()
+        for match in self.__matches:
```

Comment:
you could use a set here instead of a list, makes the code easier to read and more efficent

Reply:
Done.